test/spec/services/storage-service.js

```
@@ -128,10 +128,16 @@ describe('storageService', function () {
     expect(chromeStorageApi.get).toHaveBeenCalled();
   });
 
-  it('should be able to add new or update database table row and return promise', function () {
-    spyOn(chromeStorageApi, 'get').andReturn(deferred.promise);
-    storageService.insert('test', {'key': 'value'});
-    expect(chromeStorageApi.get).toHaveBeenCalled();
+  it('should be able to insert new database table row and return promise', function () {
```

these tests should check if on an update, that the system does not insert, and if on an insert, that it does not update (should throw an exception or return 0 rowcount if the operation is invalid)